aodntools/timeseries_products/velocity_hourly_timeseries.py

```diff
@@ -0,0 +1,414 @@
+    if 'TIME' not in variables:
+        error_list.append('TIME variable missing')
+    if 'LATITUDE' not in variables:
+        error_list.append('LATITUDE variable missing')
+    if 'LONGITUDE' not in variables:
+        error_list.append('LONGITUDE variable missing')
+    # if 'seconds_to_middle_of_measurement' not in nc['TIME'].attrs:
+    #     error_list.append('seconds_to_middle_of_measurement not present in TIME')
+
+    for variable in required_variables:
+        if variable not in variables:
+            error_list.append(variable + ' variable missing')
+        else:
+            VoIdimensions = list(nc[variable].dims)
+            if 'TIME' not in VoIdimensions:
+                error_list.append('TIME is not a dimension for ' + variable)
+            if 'LATITUDE' in VoIdimensions and len(nc.LATITUDE) > 1:
+                error_list.append('more than one LATITUDE for ' + variable)
+            if 'LONGITUDE' in VoIdimensions and len(nc.LONGITUDE) > 1:
+                error_list.append('more than one LONGITUDE for ' + variable)
+            for dim in VoIdimensions:
+                if dim not in allowed_dimensions:
+                    error_list.append('not allowed dimension: ' + dim)
+
+    if 'NOMINAL_DEPTH' not in variables and 'instrument_nominal_depth' not in attributes:
+        error_list.append('no NOMINAL_DEPTH')
+
+    return error_list
+
+def cell_velocity_resample(df, binning_function, is_WCUR):
+    """
+    Resample a dataset to a specific time_interval.
+    if WCUR not present, returns nan
+    :param df: grouped dataframe
+    :param binning_function: function used for binning as non string standard numpy function
+    :param is_WCUR: True if WCUR is present in nc, False otherwise
+    :return: binned U, v, W CUR according to the binning function
+    """
+    df_binned = df.apply(binning_function)
+    UCUR = np.array(df_binned['UCUR'])
+    VCUR = np.array(df_binned['VCUR'])
+    if is_WCUR:
+        WCUR = np.array(df_binned['WCUR'])
+    else:
+        WCUR = np.full(len(df), np.nan)
+    DEPTH = np.array(df_binned['DEPTH'])
+
+    return UCUR, VCUR, WCUR, DEPTH
+
+
+def get_resampled_values(nc_cell, ds, slice_start, varlist, binning_fun, epoch, one_day, is_WCUR):
+    """
+    get U, V, W current values resampled
+    :param nc_cell: xarray DATASET
+    :param ds: netcdf4 dataset
+    :param slice_start: start index of the slice
+    :param varlist: list of variable names to subset the dataset
+    :param binnig_fun: list of function names for binning
+    :param one_day: timedelta one day
+    :param epoch: base epoch
+    :param is_WCUR: flag indicating if WCUR is present
+    :return: end index of the slice
+    """
+    nc_cell = nc_cell.where(nc_cell.DEPTH_quality_control < 4, drop=True)
+    nc_cell = nc_cell[varlist]
+    nc_cell = nc_cell.to_dataframe()
+    ## back the index 30min
+    nc_cell.index = nc_cell.index - pd.Timedelta(30, units='m')
```

Comment:
By my experiments `Timedelta` seems to just return 30 _nanoseconds_ no matter what its `units` argument is set to! (Looks like a known issue: https://github.com/pandas-dev/pandas/issues/12136)
However, this seems to work
```suggestion
    nc_cell.index = nc_cell.index - pd.Timedelta(minutes=30)
```

Reply:
what a bug!